### PR TITLE
Stop importing harden

### PR DESCRIPTION
**Pull request by FUDCo**
A very, very large number of mostly superficial changes to eliminate importation of `@agoric/harden`, since `harden` is now a global in the SES environment.

Interestingly, testing this turned up a considerable number of tests that needed to import `@agoric/install-ses`.  These imports have been added, and various `package.json` files adjusted to reflect the many new uses of `@agoric/install-ses` and the disappearance of `@agoric/harden`. 

Since these changes intersect with a large swath of our code base, I've tagged a number of different people as reviewers, but you don't all need to review all of it, just the pieces you worry about.  I'm thinking the following:

- [x] @warner SwingSet, asssert, captp, same-structure, marshall, sparse-ints, store, swingset-runner, weak-store
- [ ] @Chris-Hibbert and/or @katelynsills ERTP, notifier, registrar, spawner, sharing-service, zoe
- [x] @michaelfig eventual-send, agoric-cli, cosmic-swingset, produce-promise
- [ ] @kriskowal bundle-source, import-manager

Of course if I've mistakenly assigned you a wrong thing, ignore it, and if you see I've assigned something to somebody else that you really should look at, look at it.  None of this is magic, just trying to spread the load.

### CODE REVIEW

**Comment by warner** [comment]
Yeah, a lot of those files were being used both from vats under swingset (where they can rely upon a `harden` global) and by non-swingset unit tests (where they must import `@agoric/harden` instead).

Once the big zoe branch lands, we'll need to rebase and update this PR; I'm sure they added a bunch of new files/tests. We might consider making these changes one package at a time, if that makes it easier to keep up with trunk.

**Review by katelynsills — APPROVED** [review]
LGTM!

**Comment by michaelfig** [comment]
One complication.  @kriskowal do we have JSDoc typing for globalThis.harden yet?  If not, we should make an issue.

**Review by michaelfig — APPROVED** [review]
LGTM!

**Comment by kriskowal** [comment]
> One complication. @kriskowal do we have JSDoc typing for globalThis.harden yet? If not, we should make an issue.

No, we don’t have TS tooling in SES yet, to the best of my knowledge.

**Review by kriskowal — APPROVED** [review]
🎉

File: packages/acorn-eventual-send/test/test-rollup.js [hunk 1]
```diff
@@ -8,7 +8,7 @@ test('SwingSet bug', async t => {
     const bundle = await rollup({
       input: require.resolve('../encouragementBotCommsWavyDot/bootstrap.js'),
       treeshake: false,
-      external: ['@agoric/evaluate', '@agoric/nat', '@agoric/harden'],
+      external: ['@agoric/evaluate', '@agoric/nat'],
```
Comment (warner) [inline packages/acorn-eventual-send/test/test-rollup.js]: This list should match the contents of `vatRequire` in `controller.js`. If `vatRequire` is now empty, this list should be too. I should have emptied it out when I landed the new-SES branch, sorry :).

File: packages/eventual-send/rollup.config.js [hunk 2]
```diff
@@ -6,9 +6,9 @@ export default [
         file: 'dist/eventual-send.umd.js',
         format: 'umd',
         name: 'EventualSend',
-        globals: {
-          '@agoric/harden': 'harden',
-        },
+        // globals: {
```
Comment (warner) [inline packages/eventual-send/rollup.config.js:9]: @michaelfig can we delete this config file now? I think I got rid of the bundling step, but missed the config file.

**Review by warner — CHANGES_REQUESTED** [review]
It looks like the XS test works by compiling two programs into C code and then running them. The first program is the equivalent of running a `tape` loader against all the `test/test*.js` files in `packages/eventual-send`:

https://github.com/Agoric/agoric-sdk/blob/36d63fcb4e0a76e8233928396e7f8a648d73b62b/.github/workflows/ag-solo-xs.yml#L71-L74

The second is the same but looking at `packages/marshal`. Both of those packages now include `test.js` files that do `install-ses`. SES includes `@agoric/babel-standalone`. That includes a regexp that looks like: 

```
  function parseSourceMapInput(str) {
    return JSON.parse(str.replace(/^\)]}'[^\n]*\n/, ''));
  }
```

which appears to [break](https://github.com/Agoric/agoric-sdk/pull/1244/checks?check_run_id=824570388#step:10:75) in XS with:

```
/home/runner/work/agoric-sdk/agoric-sdk/node_modules/@agoric/babel-standalone/babel.js:21342: error: ^\)] invalid character!
```

cc @phoddie : is there a difference between the RegExp engine used in XS and the one that we were getting from Node? I see https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/xs/XS%20Differences.md#conformance says Unicode property escapes aren't loaded by default, but that doesn't look like unicode to me. https://github.com/Moddable-OpenSource/moddable/blob/public/documentation/xs/XS%20Conformance.md#built-ins-1 lists a few RegExp tests which fail but I can't tell if any of them would cover this

@FUDCo I think the best thing to do here would be to switch these tests to use other agoric-sdk packages, which do not use `install-ses`. That way we'd still get some basic assurance that the XS makefiles still work. I'll open a separate issue to figure out the RegExp thing.

Change `.github/workflows/ag-solo-xs.yml` to use, say, `same-structure` and `weak-store` instead of `eventual-send` and `marshal`. And add a note with the regexp that XS didn't like, and a pointer here. I think that should let the CI test pass again.

All the other parts look good to me, modulo the comments below.

File: packages/SwingSet/src/vats/network/network.js [hunk 3]
```diff
@@ -1,12 +1,11 @@
+/* global harden */
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply packages/SwingSet/src/vats/network/network.js:1]: Yes, this impacts typing.  It's not an incomplete experiment, it actually worked (i.e. provided typechecking for the files).

I wrote:

> One complication. @kriskowal do we have JSDoc typing for globalThis.harden yet? If not, we should make an issue.

I will probably be the one to make the issue and implement it.

File: packages/eventual-send/rollup.config.js (hunk 2 above)
Comment (michaelfig) [reply packages/eventual-send/rollup.config.js:9]: Yes, that can be deleted.

**Comment by phoddie** [comment]
> is there a difference between the RegExp engine used in XS and the one that we were getting from Node?

The implementation in XS is independent of Node. An [introduction of our RegExp](https://blog.moddable.com/blog/regexp/) engine is available on our blog. Of course, it should be conforming to ECMA-262, except where documented.

The RegExp you note above does fail to parse with XS and succeed in Safari. We'll take a look. Thank you for the report.

**Pull request merged**